src/hsl.rs

```diff
@@ -150,15 +167,21 @@ impl<Wp, T> FromColor<Wp, T> for Hsl<Wp, T>
             hue: h.into(),
             saturation: s,
             lightness: l,
-            white_point: PhantomData,
+            space: PhantomData,
         }
     }
 
-    fn from_hsl(hsl: Hsl<Wp, T>) -> Self {
-        hsl
+    fn from_hsl<Sp: RgbSpace<WhitePoint=Wp>>(hsl: Hsl<Sp, T>) -> Self {
+        if TypeId::of::<Sp::Primaries>() == TypeId::of::<S::Primaries>() {
```

Comment:
That would need us to instantiate `Sp` and `S`, which would need `Default` or some special function that does the same thing.